### PR TITLE
Feature/sc 23829/develop ncyte statistics dashboard

### PR DESCRIPTION
**Pull request by pmanal1**
[23829](https://app.shortcut.com/clarkcan/story/23829/develop-ncyte-statistics-dashboard)

Implements the NCyTE statistics dashboard.

Notes:
- The pie charts aren't implemented in this iteration. I didn't see a route that took statistics of bloom outcomes and learning object sizes by collection.

## Demo

https://github.com/Cyber4All/clark-client/assets/93054689/1025993d-7483-4d45-97a5-4a0d53899461

Screenshot 2023-07-14 at 11 58 59 AM

[_tmp_ncyte-report-2016-01-01to2023-07-14.csv](https://github.com/Cyber4All/clark-client/files/12052030/_tmp_ncyte-report-2016-01-01to2023-07-14.csv)

### Tablet

https://github.com/Cyber4All/clark-client/assets/93054689/f65ddbec-475e-48b7-906b-0af1687e474b

### 4K

https://github.com/Cyber4All/clark-client/assets/93054689/c86d2f47-ef07-4f1b-8792-8ec537f625a2

### CODE REVIEW

**Review by bereketsega — APPROVED** [review]
looks good

**Review by mfranz1 — APPROVED** [review]

**Review by shanioffe — APPROVED** [review]
Everything looks good, nice job!

File: src/environments/environment.prod.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ export const environment = {
   contentManagerURLAdmin: 'https://api-file-upload.clark.center',
   adminAppUrl: 'https://admin.clark.center',
   utilityWebsocket: 'wss://api-utilities.clark.center',
+  clarkReportsUrl: '',
```
Comment (mfranz1) [inline src/environments/environment.prod.ts:11]: This needs the production endpint

**Review by mfranz1 — CHANGES_REQUESTED** [review]

File: src/app/collection/collection.routing.ts [hunk 2]
```diff
@@ -21,6 +22,10 @@ const collection_routes: Routes = [
         path: 'ncyte',
         component: CollectionNcyteComponent
     },
+    {
+        path: 'ncyte/dashboard',
+        component: NcyteDashboardComponent
```
Comment (mfranz1) [inline src/app/collection/collection.routing.ts:27]: Add curator guard

**Review by mfranz1 — CHANGES_REQUESTED** [review]

File: src/environments/route.ts [hunk 3]
```diff
@@ -71,6 +71,15 @@ export const COLLECTIONS_ROUTES = {
   },
   GET_COLLECTION_CURATORS(name: string ) {
     return `${environment.apiURL}/users/curators/${encodeURIComponent(name)}`;
+  },
+  GET_COLLECTION_REPORT(collections: string[], date?: { start: string, end: string }) {
+    let route = `${environment.clarkReportsUrl}?output=csv&collection=${collections.join(',')}`;
```
Comment (mfranz1) [inline src/environments/route.ts:76]: Route will need to be updated
```suggestion
    let route = `${environment.clarkReportsUrl}/reports?output=csv&collection=${collections.join(',')}`;
```

**Review by mfranz1 — CHANGES_REQUESTED** [review]

File: src/environments/environment.prod.ts (hunk 1 above)
Comment (mfranz1) [reply src/environments/environment.prod.ts:11]: @JosueMolinaMorales can you grab this pls

Comment (mfranz1) [reply src/environments/environment.prod.ts:11]: Thank you @JosueMolinaMorales

Comment (mfranz1) [reply src/environments/environment.prod.ts:11]: https://b6il5p75bqszncpwtsqbgayewi0ganvt.lambda-url.us-east-1.on.aws/

File: src/environments/environment.staging.ts [hunk 4]
```diff
@@ -7,6 +7,7 @@ export const environment = {
   contentManagerURL: 'localhost',
   contentManagerURLAdmin: 'localhost',
   adminAppUrl: 'https://admin.clark.center',
+  clarkReportsUrl: '',
```
Comment (mfranz1) [inline src/environments/environment.staging.ts:10]:
```suggestion
```

**Pull request merged**